src/System.IO/tests/BinaryWriter/BinaryWriter_WriteByteCharTests.cs

```diff
@@ -0,0 +1,573 @@
+
+            dr2.Dispose();
+            dw2.Dispose();
+            mstr.Dispose();
+        }
+
+        /// <summary>
+        /// Testing that SBytes can be written to a stream with BinaryWriter.
+        /// </summary>
+        [Fact]
+        public static void BinaryWriter_WriteSByteTest()
+        {
+            int ii = 0;
+            SByte[] sbArr = new SByte[] { 
+                SByte.MinValue, SByte.MaxValue, -100, 100, 0, SByte.MinValue / 2, SByte.MaxValue / 2, 
+                10, 20, 30, -10, -20, -30, SByte.MaxValue - 100 };
+
+            // [] read/Write with Memorystream
+            MemoryStream mstr = new MemoryStream();
+            BinaryWriter dw2 = new BinaryWriter(mstr);
+
+            for (ii = 0; ii < sbArr.Length; ii++)
+                dw2.Write(sbArr[ii]);
+
+            dw2.Flush();
+            mstr.Position = 0;
+            BinaryReader dr2 = new BinaryReader(mstr);
+
+            for (ii = 0; ii < sbArr.Length; ii++)
+            {
+                Assert.Equal(dr2.ReadSByte(), sbArr[ii]);
+            }
+
+            dr2.Dispose();
+            dw2.Dispose();
+            mstr.Dispose();
+        }
+
+        /// <summary>
+        /// Testing that an ArgumentException is thrown when Sbytes are written to a stream
+        /// and read past the end of that stream.
+        /// </summary>
+        [Fact]
+        public static void BinaryWriter_WriteSByteTest_NegativeCase()
+        {
+            int ii = 0;
+            SByte[] sbArr = new SByte[] { 
+                SByte.MinValue, SByte.MaxValue, -100, 100, 0, SByte.MinValue / 2, SByte.MaxValue / 2, 
+                10, 20, 30, -10, -20, -30, SByte.MaxValue - 100 };
+
+            MemoryStream mstr = new MemoryStream();
+            BinaryWriter dw2 = new BinaryWriter(mstr);
+
+            for (ii = 0; ii < sbArr.Length; ii++)
+                dw2.Write(sbArr[ii]);
+
+            dw2.Flush();
+
+            BinaryReader dr2 = new BinaryReader(mstr);
+            // [] Check End Of Stream
+            Assert.Throws<EndOfStreamException>(() => dr2.ReadSByte());
+
+            dr2.Dispose();
+            dw2.Dispose();
+            mstr.Dispose();
+        }
+
+        /// <summary>
+        /// Testing that a byte[] can be written to a stream with BinaryWriter.
+        /// </summary>
+        [Fact]
+        public static void BinaryWriter_WriteBArrayTest()
+        {
+            int ii = 0;
+            Byte[] bytArr = new Byte[] { Byte.MinValue, Byte.MaxValue, 1, 5, 10, 100, 200 };
+
+            // [] read/Write with Memorystream
+            MemoryStream mstr = new MemoryStream();
+            BinaryWriter dw2 = new BinaryWriter(mstr);
+
+            dw2.Write(bytArr);
+            dw2.Flush();
+            mstr.Position = 0;
+
+            BinaryReader dr2 = new BinaryReader(mstr);
+
+            for (ii = 0; ii < bytArr.Length; ii++)
+            {
+                Assert.Equal(dr2.ReadByte(), bytArr[ii]);
```

Comment:
These should be reversed such that the expected value comes first, then the actual.